drivers/dma/dma_stm32u5.c

```diff
@@ -0,0 +1,832 @@
+/*
+ * Copyright (c) 2016 Linaro Limited.
+ * Copyright (c) 2019 Song Qiang <songqiang1304521@gmail.com>
+ * Copyright (c) 2022 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Common part of DMA drivers for stm32U5.
+ * @note  Functions named with stm32_dma_* are SoCs related functions
+ *
+ */
+
+#include "dma_stm32.h"
+
+#include <zephyr/init.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/dma/dma_stm32.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(dma_stm32, CONFIG_DMA_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32u5_dma)
+#define DT_DRV_COMPAT st_stm32u5_dma
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay)
+#if DT_INST_IRQ_HAS_IDX(0, 15)
+#define DMA_STM32_0_STREAM_COUNT 16
+#elif DT_INST_IRQ_HAS_IDX(0, 7)
+#define DMA_STM32_0_STREAM_COUNT 8
+#elif DT_INST_IRQ_HAS_IDX(0, 6)
+#define DMA_STM32_0_STREAM_COUNT 7
+#elif DT_INST_IRQ_HAS_IDX(0, 5)
+#define DMA_STM32_0_STREAM_COUNT 6
+#else
+#define DMA_STM32_0_STREAM_COUNT 5
+#endif
+#endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay) */
```

Comment:
Can this be cleaned up?

Reply:
as long as stm32u5 or compatible soc series have one GPDMA of 15 channels